plutus-tx/src/PlutusTx/ErrorCodes.hs

```diff
@@ -3,12 +3,41 @@
 module PlutusTx.ErrorCodes where
 
 import           PlutusTx.Builtins as Builtins
+import           Prelude           (String)
 
 {-
 All error codes in this module should be unique and can be used only once!
 They help to trace the errors in on-chain code.
 -}
 
+{- Note [allErrorCodes]
+   This list contains all error codes used in the plutus prelude and it is
+   important that when an error code is added to the prelude it is also added
+   to this list.
+-}
+
+-- | A list of all error codes used in the plutus prelude
+allErrorCodes :: [(Builtins.BuiltinString, String)]
```

Comment:
Map?

Reply:
Good idea. Do you think the error message translations broadly make sense? We aren't actually using them in this version of the code but I'm planning to re-do this when we overhaul the QuickCheck properties in an upcoming PR.

Reply:
cc @ak3n

Reply:
>Do you think the error message translations broadly make sense?

I think that anything that makes the developer experience better worth it.